src/lib/y2users/linux/users_writer.rb

```diff
@@ -61,11 +59,14 @@ def initialize(config, initial_config, commit_configs)
       #
       # @return [Y2Issues::List] the list of issues found while writing changes; empty when none
       def write
-        with_issues do |issues|
-          issues.concat(add_users)
-          issues.concat(edit_users)
-          issues.concat(update_root_aliases)
-        end
+        @issues = Y2Issues::List.new
+
+        add_users
+        edit_users
+        # delete_users
```

Comment:
order is wrong, you have to first delete and then add/edit to prevent name/home collisions.